pylint/checkers/classes/class_checker.py

```diff
@@ -1495,11 +1495,17 @@ def _check_in_slots(self, node):
                     # Properties circumvent the slots mechanism,
                     # so we should not emit a warning for them.
                     return
-                if node.attrname in klass.locals and _has_data_descriptor(
-                    klass, node.attrname
-                ):
-                    # Descriptors circumvent the slots mechanism as well.
-                    return
+                if node.attrname in klass.locals:
+                    for local_name in klass.locals.get(node.attrname, []):
+                        statement = local_name.statement(future=True)
+                        if (
+                            isinstance(statement, nodes.AnnAssign)
+                            and not statement.value
+                        ):
+                            return
+                    if _has_data_descriptor(klass, node.attrname):
+                        # Descriptors circumvent the slots mechanism as well.
```

Comment:
```suggestion             
                for local_name in klass.locals.get(node.attrname, []):
                    statement = local_name.statement(future=True)
                    if (
                        isinstance(statement, nodes.AnnAssign)
                        and not statement.value
                    ):
                        return
                if node.attrname in klass.locals and _has_data_descriptor(klass, node.attrname):
                    # Descriptors circumvent the slots mechanism as well.
```
Or the first suggestion in the previous comment (no default value)